picard/ui/searchdialog.py

```diff
@@ -0,0 +1,466 @@
+        """Displays error inside the dialog.
+
+        Args:
+            error -- Error string
+            show_retry_button -- Whether to display retry button or not
+        """
+
+        self.error_widget = QtGui.QWidget(self)
+        self.error_widget.setObjectName("error_widget")
+        layout = QtGui.QVBoxLayout(self.error_widget)
+        error_label = QtGui.QLabel(error)
+        error_label.setWordWrap(True)
+        error_label.setAlignment(QtCore.Qt.AlignCenter)
+        error_label.setTextInteractionFlags(QtCore.Qt.TextSelectableByMouse)
+        layout.addWidget(error_label)
+        if show_retry_button:
+            retry_widget = QtGui.QWidget(self.error_widget)
+            retry_layout = QtGui.QHBoxLayout(retry_widget)
+            retry_button = QtGui.QPushButton(_("Retry"), self.error_widget)
+            retry_button.clicked.connect(self.retry)
+            retry_button.setSizePolicy(QtGui.QSizePolicy(QtGui.QSizePolicy.Maximum, QtGui.QSizePolicy.Fixed))
+            retry_layout.addWidget(retry_button)
+            retry_layout.setAlignment(QtCore.Qt.AlignHCenter | QtCore.Qt.AlignTop)
+            retry_widget.setLayout(retry_layout)
+            layout.addWidget(retry_widget)
+        self.error_widget.setLayout(layout)
+        self.add_widget_to_center_layout(self.error_widget)
+
+    def show_table(self, column_headers):
+        """Displays results table inside the dialog."""
+
+        self.table = ResultTable(self.table_headers)
+        self.table.setObjectName("results_table")
+        self.table.cellDoubleClicked.connect(self.row_double_clicked)
+        self.table.horizontalHeader().sectionResized.connect(
+                self.save_table_header_state)
+        self.restore_table_header_state()
+        self.add_widget_to_center_layout(self.table)
+        def enable_loading_button():
+            self.load_button.setEnabled(True)
+        self.table.itemSelectionChanged.connect(
+                enable_loading_button)
+
+    def row_double_clicked(self, row):
+        """Handle function for double click event inside the table."""
+
+        self.load_selection(row)
+        self.accept()
+
+    def network_error(self, reply, error):
+        error_msg = _("<strong>Following error occurred while fetching results:<br></strong>"
+                      "Network request error for %s: %s (QT code %d, HTTP code %s)<br>" % (
+                          reply.request().url().toString(QtCore.QUrl.RemoveUserInfo),
+                          reply.errorString(),
+                          error,
+                          repr(reply.attribute(QtNetwork.QNetworkRequest.HttpStatusCodeAttribute)))
+                      )
+        self.show_error(error_msg, show_retry_button=True)
+
+    def no_results_found(self):
+        error_msg = _("<strong>No results found. Please try a different search query.</strong>")
+        self.show_error(error_msg)
+
+    def accept(self):
+        if getattr(self, "table", None):
```

Comment:
To let you know which song is currently stuck in my head:

> I need to tell you something
> I really really really really really really

don't like these `getattr` calls ;-) Why not set `table` to `None` in `__init__`? It looks like the code needs to deal with `table` being `None` in any case because `add_widget_to_center_layout` can set it to that.